contracts/proto0/assets/RTokenAssetP0.sol

```diff
@@ -1,31 +1,47 @@
 // SPDX-License-Identifier: BlueOak-1.0.0
-pragma solidity 0.8.4;
+pragma solidity 0.8.9;
 
-import "../interfaces/IMain.sol";
-import "../interfaces/IVault.sol";
+import "contracts/proto0/interfaces/IMain.sol";
+import "contracts/proto0/interfaces/IVault.sol";
+import "contracts/libraries/Fixed.sol";
 import "./AssetP0.sol";
 
-// Immutable data contract, extended to implement cToken and aToken wrappers.
 contract RTokenAssetP0 is AssetP0 {
+    using FixLib for Fix;
+
     constructor(address erc20_) AssetP0(erc20_) {}
 
-    // Fiatcoins return 1e18. All redemption rates should have 18 zeroes.
-    function redemptionRate() public view override returns (uint256) {
+    function rateFiatcoin() public view override returns (Fix) {
+        assert(false);
+        return FIX_ZERO;
```

Comment:
For this, and many other cases, it'll be a bit cleaner to define errors NotImplemented and NotApplicable, and throw them from the Asset where they don't make sense.

Alternately, maybe we really do need IAsset, and ICollateral which extends IAsset, because Collateral tokens define these functions while Assets that aren't Collateral do not. That would make, e.g, calling rateFiatcoin on an RTokenAsset a compile-time error instead of a runtime error, which seems plausibly worth achieving.

Reply:
This seems worth it.